lightning-network-book.tex

```diff
@@ -240,36 +240,36 @@
+  description={Implementation of the Lightning Network Protocol by the Paris based company ACINQ (\url{https://acinq.co}). Source code is at \url{https://github.com/ACINQ/eclair}. It is written in Scala.}}
 \newglossaryentry{clightning}{name={c-lightning},
-  description={Implementation of the Lightning Network Protocol by the Victoria based Blockstream. It is written in C.}}
+  description={Implementation of the Lightning Network Protocol by the Victoria based Blockstream(\url{https://blockstream.com}). Source code is at \url{https://github.com/ElementsProject/lightning}. It is written in C.}}
 \newglossaryentry{shachain}{name={Sha-chain},
   description={}}
 \newglossaryentry{hdwallet}{name={HD Wallet},
   description={}}
 \newglossaryentry{bip}{name={BIP},
-  description={}}
+  description={BIP is short for Bitcoin Improvement Proposals. It is a mechanism by which proposals for the improvement of the Bitcoin network can be made. Submitted proposals are discussed heavily and the best one implemented. See \url{https://github.com/bitcoin/bips}.}}
 \newglossaryentry{bitcoin}{name={Bitcoin},
-  description={}}
+  description={Bitcoin is an experimental digital currency that enables payments to anyone, anywhere in the world. Bitcoin uses peer-to-peer technology to operate with no central authority: managing transactions and issuing money are carried out collectively by the network. See \url{https://www.bitcoin.org} and \url{https://github.com/bitcoin/bitcoin}.}}
 \newglossaryentry{ecash}{name={Electronic Cash},
   description={}}
 \newglossaryentry{ecdsa}{name={ECDSA},
   description={}}
 \newglossaryentry{bech32}{name={bech32},
-  description={}}
+  description={Bech32 is a segwit address format specified by BIP 0173. This address format is also known as "bc1 addresses" because all Bech32 addresses start with "bc" and currently with "bc1". See \url{https://en.bitcoin.it/wiki/Bech32}.}}
 \newglossaryentry{encoding}{name={Encoding},
-  description={}}
+  description={Encoding is the act of translating information from one form to another following specified rules. Text can be encoded as morse signals. Songs can be encoded as MP3 files. The opposite transformation is called "decoding". An MP3 song is decoded as audio.}}
 \newglossaryentry{input}{name={Input Script},
-  description={}}
+  description={The Input Script is a script to prove that you own some bitcoins.}}
 \newglossaryentry{output}{name={Output Script},
-  description={}}
+  description={The Output Script is a script where you designate someone else or multiple other parties to be able to prove they own some bitcoins.}}
 \newglossaryentry{script}{name={Bitcoin Script},
-  description={}}
+  description={Bitcoin uses a scripting system for transactions. It is called "Script" and is simple, stack-based, and processed from left to right. It is intentionally not Turing-complete, with no loops. Scripts can be embedded into Bitcoin transactions.}}
```

Comment:
they can be transacted? Isn't it a fact that every bitcoin transaction has an input and an output script by the protocol design?

Reply:
```suggestion
  description={Bitcoin uses a scripting system for transactions. It is called "Script" and is simple, stack-based, and processed from left to right. It is intentionally not Turing-complete, with no loops. Each transaction input has a script called the scriptSig, and each output has a script called the scriptPubkey. To validate an input, the input's script is concatenated with the referenced output script, and the result is executed. If the concatenation evaluates to "true", then the transaction is valid. A script is essentially a list of instructions recorded with each transaction that describe how the next person wanting to spend the Bitcoins being transferred can gain access to them.  Scripts can be written to cover a wide array of functionality with some of the best known script concepts being multisig and time-lock. A multisig script requires multiple signatures as proof before access to funds is given. A time-lock script makes funds available only after a certain time or blockchain height has been reached. The script for the simplest transfer of funds is called "pay-to-pubkey-hash" script. See \url{https://en.bitcoin.it/wiki/Script}.}}
```